vendor/VisualAcceptance.js

```diff
@@ -265,105 +292,110 @@ function utilizeImage (imageName, width, height, misMatchPercentageMargin, targe
-         ') is above mismatch threshold(' + misMatchPercentageMargin + ').')
-        }
-        data ? resolve(data) : reject(data)
-      })
-    }).then(function (data) {
-      data ? resolve(data) : reject(data)
-    }).catch(function (err) {
-      reject(err)
+        })
+      } else {
+      // Passed image exists so compare to current
+        res.image = 'data:image/png;base64,' + res.image
+        return new Promise(function (resolve, reject) {
+          resemble(res.image).compareTo(image).scaleToSameSize().onComplete(function (data) {
+            var result = false
+            if (parseFloat(data.misMatchPercentage) <= misMatchPercentageMargin) {
+            // Passed
+              $.ajax({
+                type: 'POST',
+                url: '/passed',
+                data: {
+                  image: image,
+                  name: browserDirectory + imageName + '.png'
+                }
+              })
+              result = true
+              node.innerHTML = '<div class="test pass"> <div class="list-name">  New: ' + imageName +
+              '</div> <div class="additional-info"> Addition Information: </div> <img src="' + image + '" /> </div>'
+            } else {
+            // Fail
+              $.ajax({
+                type: 'POST',
+                url: '/fail',
+                data: {
+                  image: data.getImageDataUrl(),
+                  name: browserDirectory + imageName + '.png'
+                }
+              }).then(function () {
+                node.innerHTML = '<div class="test fail"> <div class="list-name">  Changed: ' + imageName +
+                ' </div> <div class="additional-info"> Addition Information: </div> <div class="images">' +
+                '<div class="image"> <img class="diff" src="' + data.getImageDataUrl() +
+                '" /> <div class="caption">  Diff   </div> </div> <div class="image">  <img class="input" src="' +
+                image +
+                '" /> <div class="caption"> Current  </div> </div> <div class="image"> <img class="passed" src="' +
+                res.image + '" /> <div class="caption"> Baseline   </div> </div> </div> </div>'
+
+                images.push(data.getImageDataUrl())
+                images.push(image)
+                images.push(res.image)
+                $.ajax({
+                  type: 'POST',
+                  url: '/report',
+                  data: {
+                    type: 'Changed',
+                    images: images,
+                    name: imageName,
+                    browser: window.ui.browser
+                  }
+                })
+              })
+            }
+            $(document.getElementById('ember-testing')).removeAttr('style')
+            $(targetElement).removeAttr('style')
+            document.getElementsByClassName('visual-acceptance-container')[0].appendChild(node)
+            httpGet('/should-assert').then(function (response) {
```

Comment:
The same question as for `istargetbrowser`. 

Can we cache or pre-calculate `should-assert`?